core/cas-server-core-tickets-api/src/main/java/org/apereo/cas/ticket/expiration/ThrottledUseAndTimeoutExpirationPolicy.java

```diff
@@ -43,6 +44,8 @@
 
     private long timeInBetweenUsesInSeconds;
 
+    private Clock clock = Clock.systemUTC();
```

Comment:
This should be removed as a class field; it will mess with serialization of objects.

Reply:
Can we exclude this field from serialization? like Transient?

Reply:
You could, but then you'll risk null-pointer-exceptions when the object is constructed back.

Also, it doesn't make sense to add a field only for it to be used in tests. If that is case, the test should be rewritten to not require an extra field.

Reply:
Reviewed this some more, and on second thought, should be fine. Thanks!

Reply:
Wow, thank you for your understanding!  How about moving the `clock` filed into the abstract expiration policy class `AbstractCasExpirationPolicy`?  so we can write better unit tests for other child policies.

Reply:
Pleasure!

Improving tests is always a big win in my book. So, yes! That would be a welcome change and I think perhaps with that, a number of Thread.sleep() calls can also be removed. 

Keep up the good work.